packages/@react-spectrum/s2/src/Button.tsx

```diff
@@ -45,14 +45,18 @@ interface ButtonStyleProps {
   staticColor?: 'white' | 'black'
 }
 
-export interface ButtonProps extends Omit<RACButtonProps, 'className' | 'style' | 'children' | 'onHover' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, StyleProps, ButtonStyleProps {
+export interface ButtonProps extends Omit<RACButtonProps, 'className' | 'style' | 'children' | 'onHover' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, UnsafeStyles, ButtonStyleProps {
   /** The content to display in the Button. */
-  children?: ReactNode
+  children?: ReactNode,
+  /** Spectrum-defined styles, returned by the `style()` macro. */
+  styles?: StylesPropWithHeight | StylesProp
 }
 
-export interface LinkButtonProps extends Omit<LinkProps, 'className' | 'style' | 'children'>, StyleProps, ButtonStyleProps {
+export interface LinkButtonProps extends Omit<LinkProps, 'className' | 'style' | 'children'>, UnsafeStyles, ButtonStyleProps {
   /** The content to display in the Button. */
-  children?: ReactNode
+  children?: ReactNode,
+  /** Spectrum-defined styles, returned by the `style()` macro. */
+  styles?: StylesPropWithHeight | StylesProp
```

Comment:
Won't this cover it? or does this not cover width anymore?
```suggestion
  styles?: StylesPropWithHeight
```

Reply:
yep, StylesPropsWithHeight only adds height but not width: https://github.com/adobe/react-spectrum/blob/32f8b8d089056dfa7531a9ebebea15c3a56809b7/packages/%40react-spectrum/s2/src/style-utils.ts#L190-L192